test_scripts/WebEngine/commonWebEngine.lua

```diff
@@ -321,4 +321,86 @@ function common.checkUpdateAppList(pPolicyAppID, pTimes, pExpNumOfApps)
   common.wait()
 end
 
+function common.verifyPTSnapshot(appProperties, appPropExpected)
+  local snp_tbl = common.ptsTable()
+  local app_id = appProperties.policyAppID
+  local result = {}
+  local msg = ""
+
+  if (snp_tbl.policy_table.app_policies == nil) then
+    msg = msg .. "Incorrect app_policies value\n" ..
+      " Expected: exists \n" ..
+      " Actual: nil \n"
+  end
+
+  if (snp_tbl.policy_table.consumer_friendly_messages == nil) then
+    msg = msg .. "Incorrect consumer_friendly_messages value\n" ..
+      " Expected: exists \n" ..
+      " Actual: nil \n"
+  end
+
+  if (snp_tbl.policy_table.device_data == nil) then
+    msg = msg .. "Incorrect device_data value\n" ..
+      " Expected: exists \n" ..
+      " Actual: nil \n"
+  end
+
+  if (snp_tbl.policy_table.functional_groupings == nil) then
+    msg = msg .. "Incorrect functional_groupings value\n" ..
+      " Expected: exists \n" ..
+      " Actual: nil \n"
+  end
+
+  if (snp_tbl.policy_table.module_config == nil) then
+    msg = msg .. "Incorrect module_config value\n" ..
+      " Expected: exists \n" ..
+      " Actual: nil \n"
+  end
+
+  if (snp_tbl.policy_table.usage_and_error_counts == nil) then
+    msg = msg .. "Incorrect usage_and_error_counts value\n" ..
+      " Expected: exists \n" ..
+      " Actual: nil \n"
+  end  
+
+  local nicknames = snp_tbl.policy_table.app_policies[app_id].nicknames
+  if not common.isTableEqual(nicknames, appPropExpected.nicknames) then
+    msg = msg .. "Incorrect nicknames\n" ..
+      " Expected: " .. common.tableToString(appPropExpected.nicknames) .. "\n" ..
+      " Actual: " .. common.tableToString(nicknames) .. "\n"
+  end
+
+  local auth_token = snp_tbl.policy_table.app_policies[app_id].auth_token
+  if not (auth_token == appPropExpected.auth_token) then
+    msg = msg .. "Incorrect auth token value\n" ..
+      " Expected: " .. appPropExpected.auth_token .. "\n" ..
+      " Actual: " .. auth_token .. "\n"
+  end
+
+  local cloud_transport_type = snp_tbl.policy_table.app_policies[app_id].cloud_transport_type
+  if not (cloud_transport_type == appPropExpected.cloud_transport_type) then
+    msg = msg ..     "Incorrect cloud_transport_type value\n" ..
+      " Expected: " .. appPropExpected.cloud_transport_type .. "\n" ..
+      " Actual: " .. cloud_transport_type .. "\n"
+  end
+
+  local enabled = tostring(snp_tbl.policy_table.app_policies[app_id].enabled)
```

Comment:
```suggestion
  local enabled = snp_tbl.policy_table.app_policies[app_id].enabled
```

requesting one last change: `enabled` in the appPropsExpected should just be a bool, not a string. It will make using this function in the future more intuitive (see how appProps has a bool enabled)